### PR TITLE
[test] Fix workspace integration tests

### PR DESCRIPTION
**Pull request by kylos101**
## Description

Stabilize workspace integration tests (we're down to one failure now, TestGitActions, which intermittently fails), which we'll tackle in a separate PR.

This required that we:
1. update the packer image to the version being used in gen95 (WKS-107). The image contained many fixes (including a bump to containerd 1.6.20, K3s 1.26, etc.). These updates help avoid disk pressure issues, which was causing tests to fail intermittently. :bulb:  I suspect this was also leading to preview environments being unstable in general.
2. make code changes in `/test` to account for recent product changes (like `organizationId` being required for `createWorkspace` with `server`)
3. leverage the `trust-manager` manifests built into the packer image, and modify the startup script, so that it retries applying `trust-manager` manifests to workaround timing issues causing preview environment failures.

Also, I've added `/test` sub-folders to `.github/CODEOWNERS`. This should help teams (1) inspect changes (2) assert tests pass before merge to main and (3) collaborate on testing.

## Related 

Fixes WKS-70
Fixes WKS-107
Fixes PDEO-7
https://github.com/gitpod-io/gitpod/pull/17335

## How to test

1. `git checkout -b some-name` to a local branch, and then build a preview `TF_VAR_infra_provider="gce" TF_VAR_with_large_vm=true leeway run dev:preview` (Why? The branch for this PR is rate limited until May 2 at 6pm EST via Let's Encrypt.)
2. run the [workspace integration tests](https://github.com/gitpod-io/gitpod/actions/workflows/workspace-integration-tests.yml) from `kylos101/gen94-integ` branch, with asset `kylos101-gen94-integ-gha.8496`. Here's a sample [run](https://github.com/gitpod-io/gitpod/actions/runs/4834678030) that passed :heavy_check_mark: .

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

**Comment by kylos101** [comment]
@mustard-mh and @akosyakov  for :eyes: , because I'm changing integration tests (workspace tests have been horribly broken), and adding `/test` sub-folders to CODEOWNERS. Will hopefully be marking this PR as ready for review soon. :pray:

**Comment by akosyakov** [comment]
On our side if running `ide` as target works then it is alright. When PR is not draft we are happy to review. @iQQBot looked quite a lot in integration testing, for sure can help 🙏

File: .github/CODEOWNERS
Comment (kylos101) [reply .github/CODEOWNERS]: Thank you! [Fixed](https://github.com/gitpod-io/gitpod/compare/23ce7405d72c30cb24df95adcf71a6c527d9ddcf..61fa2733186f6c2a39003ef6d02539e97ba833b8).

File: test/README.md
Comment (kylos101) [reply test/README.md:71]: This, was not my experience, but, I'll try again to confirm. Thanks for the thorough review, @iQQBot ! :hugs:

**Comment by kylos101** [comment]
The `trust-manager` fix in action:
```
root@kylos101-gc38a6d66fe:/home/kyle# journalctl -t install-trustmanager -xf
Apr 28 18:51:26 kylos101-gc38a6d66fe install-trustmanager[3784]: Starting to install trust manager
Apr 28 18:51:26 kylos101-gc38a6d66fe install-trustmanager[3818]: Sleeping...
Apr 28 18:51:32 kylos101-gc38a6d66fe install-trustmanager[4988]: Trust manager applied
Apr 28 18:51:32 kylos101-gc38a6d66fe install-trustmanager[5039]: Sleeping...
Apr 28 18:51:38 kylos101-gc38a6d66fe install-trustmanager[5929]: Trust manager applied
Apr 28 18:51:38 kylos101-gc38a6d66fe install-trustmanager[5955]: Sleeping...
Apr 28 18:51:43 kylos101-gc38a6d66fe install-trustmanager[6497]: Trust manager applied
Apr 28 18:51:43 kylos101-gc38a6d66fe install-trustmanager[6523]: Sleeping...
Apr 28 18:51:48 kylos101-gc38a6d66fe install-trustmanager[7101]: Trust manager applied
Apr 28 18:52:11 kylos101-gc38a6d66fe install-trustmanager[8364]: Finishing installing trust manager
```

**Comment by kylos101** [comment]
Can build preview envs:
```
TF_VAR_infra_provider="gce" TF_VAR_with_large_vm=true leeway run dev:preview
SUCCESS: Installation is happy: https://kylos101-gc38a6d66fe.preview.gitpod-dev.com/workspaces
FATA[0575] failed with exit code 1                      
FATA[0662] failed with exit code 1   
```

image

**Comment by kylos101** [comment]
Can build preview envs:
```
TF_VAR_infra_provider="gce" TF_VAR_with_large_vm=true leeway run dev:preview
SUCCESS: Installation is happy: https://kylos101-gc38a6d66fe.preview.gitpod-dev.com/workspaces
```

image

File: dev/preview/infrastructure/scripts/install-trustmanager.sh
Comment (kylos101) [inline dev/preview/infrastructure/scripts/install-trustmanager.sh:1]: Fixes PDEO-7

File: dev/preview/infrastructure/modules/gce/variables.tf
Comment (kylos101) [inline dev/preview/infrastructure/modules/gce/variables.tf:47]: WKS-107

File: dev/preview/BUILD.yaml
Comment (kylos101) [inline dev/preview/BUILD.yaml:36]: WKS-107

File: dev/preview/infrastructure/modules/harvester/variables.tf
Comment (kylos101) [inline dev/preview/infrastructure/modules/harvester/variables.tf:36]: WKS-107

File: dev/preview/infrastructure/variables.tf
Comment (kylos101) [inline dev/preview/infrastructure/variables.tf:38]: WKS-107

File: dev/preview/infrastructure/modules/gce/vm.tf
Comment (kylos101) [inline dev/preview/infrastructure/modules/gce/vm.tf:80]: Fixes PDEO-7

File: dev/preview/infrastructure/scripts/install-trustmanager.sh
Comment (kylos101) [inline dev/preview/infrastructure/scripts/install-trustmanager.sh:1]: Fixes PDEO-7

File: dev/preview/workflow/preview/deploy-gitpod.sh
Comment (kylos101) [inline dev/preview/workflow/preview/deploy-gitpod.sh:1]: Cleanup for PDEO-7

File: dev/preview/workflow/preview/deploy-monitoring-satellite.sh
Comment (kylos101) [inline dev/preview/workflow/preview/deploy-monitoring-satellite.sh:78]: A byproduct of no longer supporting PSP after WKS-107

**Review by kylos101 — COMMENTED** [review]
Self review

File: test/README.md
Comment (kylos101) [reply test/README.md:71]: Works! Will update.

```
gitpod /workspace/gitpod/test (test-me) $ export USER_NAME=kylos101
gitpod /workspace/gitpod/test (test-me) $ export USER_TOKEN=foo
gitpod /workspace/gitpod/test (test-me) $ go test -v ./... \
    -kubeconfig=/home/gitpod/.kube/config \
    -namespace=default \
    -run TestWorkspaceInstrumentation
?       github.com/gitpod-io/gitpod/test/pkg/agent/daemon       [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/daemon/api   [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/workspace    [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api        [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/integration        [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/integration/common [no test files]
?       github.com/gitpod-io/gitpod/test/pkg/report     [no test files]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/content-service       4.309s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/database      4.157s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/image-builder 4.245s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/server        4.175s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/ws-daemon     4.284s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/components/ws-manager    4.293s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/ide/jetbrains    4.296s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/ide/ssh  4.332s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/ide/vscode       4.153s [no tests to run]
testing: warning: no tests to run
PASS
ok      github.com/gitpod-io/gitpod/test/tests/smoke-test       4.314s [no tests to run]
=== RUN   TestWorkspaceInstrumentation
=== RUN   TestWorkspaceInstrumentation/instrumentation
=== RUN   TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace
=== RUN   TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace/https://github.com/gitpod-io/empty
    report.go:11: example
    report.go:13: this is the example test for instrumenting a workspace
=== PAUSE TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace/https://github.com/gitpod-io/empty
=== CONT  TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace/https://github.com/gitpod-io/empty
    workspace.go:369: attempt to create the workspace as user 4a819e95-0b79-45dc-a73f-08549cf514a6, with context https://github.com/gitpod-io/empty
    workspace.go:411: attempt to get the workspace information: gitpodio-empty-sgl94ck15qf
    workspace.go:423: not preparing
    workspace.go:432: got the workspace information: gitpodio-empty-sgl94ck15qf
    workspace.go:460: wait for workspace to be fully up and running
    workspace.go:569: prepare for a connection with ws-manager
    workspace.go:590: established for a connection with ws-manager
    workspace.go:598: check if the status of workspace is in the running phase: b7e1932f-1d19-4834-a146-5e861932884b
    workspace.go:631: status: b7e1932f-1d19-4834-a146-5e861932884b, INITIALIZING
    workspace.go:598: check if the status of workspace is in the running phase: b7e1932f-1d19-4834-a146-5e861932884b
    workspace.go:631: status: b7e1932f-1d19-4834-a146-5e861932884b, INITIALIZING
    workspace.go:598: check if the status of workspace is in the running phase: b7e1932f-1d19-4834-a146-5e861932884b
    workspace.go:631: status: b7e1932f-1d19-4834-a146-5e861932884b, RUNNING
    workspace.go:655: confirmed that the worksapce is running: b7e1932f-1d19-4834-a146-5e861932884b, RUNNING
    workspace.go:468: successful launch of the workspace
    example_test.go:93: .git
    workspace.go:504: waiting for stopping the workspace: b7e1932f-1d19-4834-a146-5e861932884b
    workspace.go:514: attemp to delete the workspace: b7e1932f-1d19-4834-a146-5e861932884b
    workspace.go:797: confirmed the worksapce is stopped: b7e1932f-1d19-4834-a146-5e861932884b, STOPPED
    workspace.go:538: successfully terminated workspace
--- PASS: TestWorkspaceInstrumentation (143.71s)
    --- PASS: TestWorkspaceInstrumentation/instrumentation (143.71s)
        --- PASS: TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace (0.00s)
            --- PASS: TestWorkspaceInstrumentation/instrumentation/it_can_instrument_a_workspace/https://github.com/gitpod-io/empty (143.71s)
PASS
ok      github.com/gitpod-io/gitpod/test/tests/workspace        148.008s
```

Comment (kylos101) [reply test/README.md]: Fixed in 06fc4f9794f2442ed6cdfb7726b701e9e06d2fe7

**Review by WVerlaek — COMMENTED** [review]
👍 

only some nits, lgtm otherwise

File: .werft/vm/manifests/rook-ceph/common.yaml
Comment (kylos101) [reply .werft/vm/manifests/rook-ceph/common.yaml:100]: This was a miss, good :eyes: , will remove

Comment (kylos101) [reply .werft/vm/manifests/rook-ceph/common.yaml]: Removed in [44f24fb](https://github.com/gitpod-io/gitpod/pull/17222/commits/44f24fb0312bd3c3e7b7b7feaf39916462e21c18)

File: components/gitpod-protocol/go/gitpod-service.go
Comment (kylos101) [reply components/gitpod-protocol/go/gitpod-service.go]: Removed in [44f24fb](https://github.com/gitpod-io/gitpod/pull/17222/commits/44f24fb0312bd3c3e7b7b7feaf39916462e21c18)

File: .werft/vm/manifests/rook-ceph/common.yaml
Comment (WVerlaek) [reply .werft/vm/manifests/rook-ceph/common.yaml]: Thanks - there's also some other commented out blocks in this file

**Review by WVerlaek — APPROVED** [review]
lgtm after last comment is resolved

**Comment by kylos101** [comment]
> b9c9a2e

Found a bit more in [b9c9a2e](https://github.com/gitpod-io/gitpod/pull/17222/commits/b9c9a2eff98bae56412415fe40c889be106087cd), but, that should be it

**Comment by kylos101** [comment]
> lgtm after last comment is resolved

Found a bit more in [b9c9a2e](https://github.com/gitpod-io/gitpod/pull/17222/commits/b9c9a2eff98bae56412415fe40c889be106087cd), but, that should be it.

**Review by easyCZ — APPROVED** [review]

**Comment by kylos101** [comment]
@gitpod-io/engineering-ide hey there, could I trouble you to trigger your integration tests from this branch? I ask to assert your tests are not broken with any related changes. :pray:

**Comment by kylos101** [comment]
Tested we can build harvester preview environments in a separate branch and PR - https://github.com/gitpod-io/gitpod/actions/runs/4865460534 & https://github.com/gitpod-io/gitpod/pull/17469, removing the hold, as IDE tests pass too.

**Comment by kylos101** [comment]
> Let's continue the hold. Of course cloudinit fails for harvester preview VMs.

It doesn't make sense to support customizations for the GCP packer image for both gce and harvester. To support harvester we have to fix (1) [fix calico customizations for containerd](https://github.com/gitpod-io/gitpod-packer-gcp-image/pull/231) :heavy_check_mark:  (2) redo the trust-manager installation as part of the cloudinit.yml script for harvester :heavy_check_mark:  (3) get llvm working somehow in harvester :x: and (4) who knows what else.

Let's try to force all preview environments to gce in https://github.com/gitpod-io/gitpod/pull/17475 in support of PDEO-2.

**Comment by vulkoingim** [comment]
Btw. I think I might have mentioned this before - or maybe I haven't 🤔  But you can run your integration tests with [spot VMs](https://github.com/gitpod-io/gitpod/blob/main/dev/preview/infrastructure/preview.tf#LL9C3-L9C3) and also pass a different [VM type for GCE](https://github.com/gitpod-io/gitpod/blob/main/dev/preview/infrastructure/preview.tf#L11)

That will make it much cheaper, and also allow you to use much larger VMs, which in theory means that they'll run also faster (not sure if there's a resource bottleneck right now). E.g. you can pass `vm_type=c2-standard-30` , without `with_large_vm=true`, you'll get that machine type.

**Comment by kylos101** [comment]
One test intermittently failed [here](https://github.com/gitpod-io/gitpod/actions/runs/4875582597): TestOpenWorkspaceFromOutdatedPrebuild/prebuild/it_should_open_a_workspace_from_with_an_older_prebuild_initializer_successfully_and_run_the_init_task/classic: github.com/gitpod-io/gitpod/test/tests/components/ws-manager

Will follow-up in a separate PR.

**Pull request merged**
